awscli/customizations/cloudfront.py

```diff
@@ -58,6 +70,44 @@ def add_to_params(self, parameters, value):
                 }
 
 
+def _add_default_root_object(argument_table, **kwargs):
+    argument_table['distribution-config'].required = False
+    argument_table['default-root-object'] = DefaultRootObjectArgument(
+        kwargs['session'])
+
+
+class DefaultRootObjectArgument(CustomArgument):
+
+    def __init__(self, session):
+        doc = (
+            'The object that you want CloudFront to return (for example, '
+            'index.html) when a viewer request points to your root URL. '
+            'Note: --default-root-object '
+            'and --distribution-config are mututally exclusive.'
+        )
+        super(DefaultRootObjectArgument, self).__init__(
+            'default-root-object', help_text=doc)
+        self.session = session
+
+    def add_to_params(self, parameters, value):
+        if value is not None:
+            parameters[self.name] = value
+
+
+def _update_distribution(parsed_globals, call_parameters, **kwargs):
+    if ('default-root-object' in call_parameters
```

Comment:
Why would this be in the ``call_parameters``?  It should only be in the ``parsed_args``. We should not be putting parameters into the call_parameters in the first place that has no chance to be used by the client.